### PR TITLE
Add sso_enabled flag to user model

### PR DESCRIPTION
**Pull request by knolleary**
Part of #226 

 - This adds a new flag to the User model, `sso_enabled` that indicates this user has logged in via SSO.

 - If the flag is true for a given user, they cannot edit their email address.

image

 - Adds an sso_enabled column to the admin table of users

image

With this PR, nothing sets this property to true. That'll come in the next PR that introduces SSO login.

### CODE REVIEW

**Review by hardillb — APPROVED** [review]

File: forge/db/views/User.js [hunk 1]
```diff
@@ -46,6 +46,9 @@ module.exports = {
         if (user.defaultTeamId) {
             result.defaultTeam = app.db.models.Team.encodeHashid(user.defaultTeamId)
         }
+        if (app.config.features.enabled('sso')) {
+            result.sso_enabled = !!user.sso_enabled
```
Comment (hardillb) [inline forge/db/views/User.js:50]: Only sets the property if true. Test checks it's false

File: test/unit/forge/routes/api/user_spec.js [hunk 2]
```diff
@@ -101,7 +105,36 @@ describe('User API', async function () {
...
+                method: 'GET',
+                url: '/api/v1/user',
+                cookies: { sid: TestObjects.tokens.alice }
+            })
+            response.statusCode.should.equal(200)
+            const result = response.json()
+            result.should.have.property('id', TestObjects.alice.hashid)
+            result.should.have.property('username', TestObjects.alice.username)
+            result.should.have.property('email', TestObjects.alice.email)
+            result.should.have.property('sso_enabled', false)
```
Comment (hardillb) [inline test/unit/forge/routes/api/user_spec.js:123]: This fails because the view only adds the SSO field if it's set to true

**Review by hardillb — CHANGES_REQUESTED** [review]

File: test/unit/forge/routes/api/user_spec.js (hunk 2 above)
Comment (knolleary) [reply test/unit/forge/routes/api/user_spec.js:123]: Odd - tests passes locally. Will investigate.

**Comment by knolleary** [comment]
Have pushed a bit more to the branch to enable the missing bits.

**Review by hardillb — APPROVED** [review]

**Pull request merged**